src/number_theory/padics/padic_integers.lean

```diff
@@ -111,16 +111,20 @@ instance : has_one ℤ_[p] :=
 
 @[simp, norm_cast] lemma coe_one : ((1 : ℤ_[p]) : ℚ_[p]) = 1 := rfl
 
-@[simp, norm_cast] lemma coe_coe : ∀ n : ℕ, ((n : ℤ_[p]) : ℚ_[p]) = n
-| 0 := rfl
-| (k+1) := by simp [coe_coe]
-
+@[simp, norm_cast] lemma coe_zero : ((0 : ℤ_[p]) : ℚ_[p]) = 0 := rfl
 
-@[simp, norm_cast] lemma coe_coe_int : ∀ (z : ℤ), ((z : ℤ_[p]) : ℚ_[p]) = z
-| (int.of_nat n) := by simp
-| -[1+n] := by simp
+instance : add_comm_group ℤ_[p] :=
+by refine_struct
+{ add   := (+),
+  neg   := has_neg.neg,
+  zero  := (0 : ℤ_[p]),
+  sub   := has_sub.sub,
+  nsmul := @nsmul_rec _ ⟨(0 : ℤ_[p])⟩ ⟨(+)⟩,
+  zsmul := @zsmul_rec _ ⟨(0 : ℤ_[p])⟩ ⟨(+)⟩ ⟨has_neg.neg⟩ };
+intros; try { refl }; ext; simp; ring
 
-@[simp, norm_cast] lemma coe_zero : ((0 : ℤ_[p]) : ℚ_[p]) = 0 := rfl
+instance : add_group_with_one ℤ_[p] :=
+{ one := 1, .. padic_int.add_comm_group }
```

Comment:
```suggestion
{ one := 1, .. padic_int.add_comm_group }
```
Can you define the `cast` more nicely so that `coe_coe` and `coe_coe_int` are definitional, or make a comment with this as todo?